packages/slider/src/HandleController.ts

```diff
@@ -0,0 +1,540 @@
+            'background-color': `var(--spectrum-slider-handle-background-color-${index}, var(--spectrum-slider-handle-default-background-color))`,
+            'border-color': `var(--spectrum-slider-handle-border-color-${index}, var(-spectrum-slider-handle-default-border-color))`,
+        };
+        return html`
+            <div
+                class=${classMap(classes)}
+                name=${model.name}
+                style=${styleMap(style)}
+                @manage=${streamingListener(
+                    {
+                        type: 'pointerdown',
+                        fn: (event) => this.handlePointerdown(event, model),
+                    },
+                    {
+                        type: 'pointermove',
+                        fn: (event) => this.handlePointermove(event, model),
+                    },
+                    {
+                        type: ['pointerup', 'pointercancel'],
+                        fn: (event) => this.handlePointerup(event, model),
+                    }
+                )}
+                role="presentation"
+            >
+                <input
+                    type="range"
+                    class="input"
+                    id=${ifDefined(isActive ? 'input' : undefined)}
+                    min=${model.clamp.min}
+                    max=${model.clamp.max}
+                    step=${model.step}
+                    value=${model.value}
+                    aria-disabled=${ifDefined(
+                        this.host.disabled ? 'true' : undefined
+                    )}
+                    aria-valuetext=${this.formattedValueForHandle(model)}
+                    @change=${(event: Event) =>
+                        this.onInputChange(event, model)}
+                    @focus=${(event: Event) => this.onInputFocus(event, model)}
+                    @blur=${() => this.onInputBlur(model)}
+                />
+            </div>
+        `;
+    }
+
+    public render(): TemplateResult[] {
+        this.clearHandleComponentCache();
+        return this.model.map((model, index) => {
+            const zIndex = this.handleOrder.indexOf(model.name) + 1;
+            return this.renderHandle(model, index, zIndex);
+        });
+    }
+
+    /**
+     * Returns a list of track segment [start, end] tuples where the values are
+     * normalized to be between 0 and 1.
+     * @returns A list of track segment tuples [start, end]
+     */
+    public trackSegments(): [number, number][] {
+        const values = this.model.map((model) => model.normalizedValue);
+        values.sort((a, b) => a - b);
+
+        // The first segment always starts at 0
+        values.unshift(0);
+        return values.map((value, index, array) => [
+            value,
+            array[index + 1] ?? 1,
+        ]);
+    }
+
+    private updateModel(): void {
+        const handles = [...this.handles.values()];
+
+        const getRangeAndClamp = (index: number): RangeAndClamp => {
```

Comment:
This is a pretty smart function. Almost enough to want to have it moved somewhere it could be individually unit tested. Not a blocker.